File: backend/src/main/java/bingol/campus/student/core/response/HomeStoryDTO.java
package bingol.campus.student.core.response;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;
import java.util.UUID;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class HomeStoryDTO {
    private List<UUID> storyId;
    private long studentId;
    private String username;
    private List<String>photos;
    private String profilePhoto;
    private boolean isVisited;

}

File: frontend/lib/models/home_story_dto.dart
import 'package:json_annotation/json_annotation.dart';

part 'home_story_dto.g.dart';

@JsonSerializable()
class HomeStoryDTO {
  final List<String> storyId; // UUID'ler String olarak temsil edilir
  final int studentId; // long türü Dart'ta int olarak temsil edilir
  final String username; // Kullanıcı adı
  final List<String> photos; // Fotoğraflar
  final String profilePhoto; // Profil fotoğrafı URL'si
  @JsonKey(name: 'visited') // Map 'visited' from API to 'isVisited' in class
  final bool isVisited; // Ziyaret edildi mi?

  HomeStoryDTO({
    required this.storyId,
    required this.studentId,
    required this.username,
    required this.photos,
    required this.profilePhoto,
    required this.isVisited,
  });

  factory HomeStoryDTO.fromJson(Map<String, dynamic> json) => _$HomeStoryDTOFromJson(json);
  Map<String, dynamic> toJson() => _$HomeStoryDTOToJson(this);
} 